Dangerfile

```diff
@@ -209,12 +217,12 @@ module VSPDanger
 end
 
 if $PROGRAM_NAME != __FILE__
-  VSPDanger::Runner.run.each do |output|
-    case output[:severity]
-    when :error
-      fail output[:message] # rubocop:disable Style/SignalException
-    when :warning
-      warn output[:message]
+  VSPDanger::Runner.run.each do |result|
+    case result.severity
+    when VSPDanger::Result::ERROR
+      failure result.message
+    when VSPDanger::Result::WARNING
+      warn result.message
```

Comment:
The logic doesn't seem to be different than before.

Is the real fix here using `failure` over `fail`?

(I'm fine with the Result refactor.)

Reply:
Yeah... It was, but then I backtracked the more I read about how Danger handles failure and warnings

Reply:
At one point it was this
```
results = VSPDanger::Runner.run

  errors = results.select { |result| result.severity == VSPDanger::Result::ERROR }
  if errors.any?
    fail(errors.collect(&:message)) # rubocop:disable Style/SignalException
  end

  warnings = results.select { |result| result.severity == VSPDanger::Result::WARNING }
  if warnings.any? # rubocop:disable Style/IfUnlessModifier
    warn(warnings.collect(&:message))
  end
  ```
  But I assumed fail and warn were terminal, but they are not